### PR TITLE
docs(textarea): migration guide and v7 playgrounds

### PR DESCRIPTION
**Pull request by sean-perkins**
- Updates the dev-build to a new build off the latest `feature-7.0` changes
- Adds the migration guide for the modern textarea rendering (following the template from `ion-input`)

Adds new component playgrounds for:
- Label placement
- Fill/Outline styles
- Helper/error text
- Counter formatter

Quick preview: https://ionic-docs-git-docs-v7-textarea-ionic1.vercel.app/docs/api/textarea

Note: Angular and Javascript (HTML) playground examples are still effected by the Stackblitz bug with Jsdelivr/loading v6.

### CODE REVIEW

File: docs/api/textarea.md [hunk 1]
```diff
@@ -29,6 +29,40 @@ import BasicPlayground from '@site/static/usage/v7/textarea/basic/index.md';
...
+
+<Fill />
+
+## Helper & Error Text
+
+Helper and error text can be used inside of a textarea with the `helperText` and `errorText` property. The error text will not be displayed unless the `ion-invalid` class is added to the `ion-textarea`. In Angular, this is done automatically through form validation. In JavaScript, React and Vue, the class needs to be manually added based on your own validation.
+
+import HelperError from '@site/static/usage/v7/textarea/helper-error/index.md';
+
+<HelperError />
```
Comment (amandaejohnston) [inline docs/api/textarea.md:56]: The React example doesn't seem to be working in Stackblitz; I don't see the helper or error text.

File: docs/api/textarea.md [hunk 2]
```diff
@@ -29,6 +29,40 @@ import BasicPlayground from '@site/static/usage/v7/textarea/basic/index.md';
...
+
+## Textarea Counter
+
+The textarea counter is text that displays under a textarea to notify the user of how many characters have been entered out of the total that the textarea will accept. When adding counter, the default behavior is to format the value that gets displayed as `inputLength` / `maxLength`. This behavior can be customized by passing in a formatter function to the `counterFormatter` property.
+
+import Counter from '@site/static/usage/v7/textarea/counter/index.md';
+
+<Counter />
```
Comment (amandaejohnston) [inline docs/api/textarea.md:64]: Could we make this playground a little taller so there's no scroll bar?

Comment (amandaejohnston) [inline docs/api/textarea.md:64]: I don't see the bottom content when opening the React Stackblitz here either; I wonder if that's just an Ionic bug?

File: static/usage/v7/textarea/migration/index.md [hunk 3]
```diff
@@ -0,0 +1,224 @@
...
+  <ion-label position="floating">Label:</ion-label>
+  <ion-textarea></ion-textarea>
+</ion-item>
+
+<!-- After -->
+
+<!-- Textareas using `fill` should not be placed in ion-item -->
+<ion-textarea fill="outline" shape="round" label="Label:" labelPlacement="floating"></ion-textarea>
+
+<!-- Input-specific features on ion-item -->
```
Comment (amandaejohnston) [inline static/usage/v7/textarea/migration/index.md:97]:
```suggestion
<!-- Textarea-specific features on ion-item -->
```

File: static/usage/v7/textarea/migration/index.md [hunk 4]
```diff
@@ -0,0 +1,224 @@
...
+  <IonLabel position="floating">Label:</IonLabel>
+  <IonTextarea></IonTextarea>
+</IonItem>
+
+{/* After */}
+
+{/* Inputs using `fill` should not be placed in IonItem */}
+<IonTextarea fill="outline" shape="round" label="Label:" labelPlacement="floating"></IonTextarea>
+
+{/* Input-specific features on IonItem */}
```
Comment (amandaejohnston) [inline static/usage/v7/textarea/migration/index.md:149]:
```suggestion
{/* Textarea-specific features on IonItem */}
```

**Review by amandaejohnston — CHANGES_REQUESTED** [review]

File: docs/api/textarea.md (hunk 1 above)
Comment (sean-perkins) [reply docs/api/textarea.md:56]: This looks like a bug, I need to dig it more. It is rendering, but the textarea is filing the entire `ionContent` space 😆 so it pushes the slots off the visual viewport. 

I don't think the textarea should fill the container space like this..

File: docs/api/textarea.md [hunk 5]
```diff
@@ -45,6 +79,29 @@ import ClearOnEditPlayground from '@site/static/usage/v7/textarea/clear-on-edit/
 
 <ClearOnEditPlayground />
 
+## Migrating from Legacy Textarea Syntax
+
+A simpler textarea syntax was introduced in Ionic 7.0. This new syntax reduces the boilerplate required to setup an textarea, resolves accessibility issues, and improves the developer experience.
+
+While developers can continue using the legacy syntax, we recommend migrating as soon as possible.
```
Comment (liamdebeasi) [inline docs/api/textarea.md:86]:
```suggestion
Developers can perform this migration one textarea at a time. While developers can continue using the legacy syntax, we recommend migrating as soon as possible.
```

**Review by liamdebeasi — CHANGES_REQUESTED** [review]

File: docs/api/textarea.md [hunk 6]
```diff
@@ -29,6 +29,40 @@ import BasicPlayground from '@site/static/usage/v7/textarea/basic/index.md';
 
 <BasicPlayground />
 
+## Label Positioning
```
Comment (liamdebeasi) [inline docs/api/textarea.md:32]: Can we rename this "Label Placement"? I'd like to converge on a single naming (See https://github.com/ionic-team/ionic-docs/pull/2747)

File: docs/api/textarea.md [hunk 7]
```diff
@@ -29,6 +29,40 @@ import BasicPlayground from '@site/static/usage/v7/textarea/basic/index.md';
...
+
+Labels will take up the width of their content by default. This positioning can be changed to be a fixed width, stacked, or floating label.
+
+import Labels from '@site/static/usage/v7/textarea/labels/index.md';
+
+<Labels />
```
Comment (liamdebeasi) [inline docs/api/textarea.md:38]: Can we rename the directory `label-placement`? (See previous comment on consistency)

File: docs/api/textarea.md [hunk 8]
```diff
@@ -29,6 +29,40 @@ import BasicPlayground from '@site/static/usage/v7/textarea/basic/index.md';
 
 <BasicPlayground />
 
+## Label Positioning
+
+Labels will take up the width of their content by default. This positioning can be changed to be a fixed width, stacked, or floating label.
```
Comment (liamdebeasi) [inline docs/api/textarea.md:34]:
```suggestion
Labels will take up the width of their content by default. Developers can use the `labelPlacement` property to control how the label is placed relative to the control.
```

**Review by liamdebeasi — CHANGES_REQUESTED** [review]

**Comment by sean-perkins** [comment]
I'm going to update the `ion-textarea` implementation to have the wrapper inherit the height (this aligns with `ion-input`): https://github.com/ionic-team/ionic-framework/pull/26707.

Still getting inconsistent results in Stackblitz and at least changing the styles in the dev-tools results in the correct rendering. I'll also verify with a dev-build.

**Comment by sean-perkins** [comment]
The issues with the bottom content not displaying in the React playground have been resolved.

**Review by amandaejohnston — APPROVED** [review]

**Review by liamdebeasi — APPROVED** [review]

**Pull request merged**